server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -140,14 +141,16 @@ public static boolean isDedicatedFrozenNode(final Settings settings) {
 
     private static final StringLiteralDeduplicator nodeStringDeduplicator = new StringLiteralDeduplicator();
 
+    public record VersionInformation(Version nodeVersion, IndexVersion minIndexVersion, IndexVersion maxIndexVersion) {}
```

Comment:
Should the min index version always be specified, or should we infer it from semver on IndexVersion using the major version number?